libmemcached/storage.cc

```diff
@@ -432,7 +445,21 @@ memcached_return_t memcached_set(memcached_st *ptr, const char *key, size_t key_
   memcached_return_t rc;
   LIBMEMCACHED_MEMCACHED_SET_START();
   rc= memcached_send(ptr, key, key_length,
-                     key, key_length, value, value_length,
+                     key, key_length, 0, value, value_length,
+                     expiration, flags, 0, SET_OP);
+  LIBMEMCACHED_MEMCACHED_SET_END();
+  return rc;
+}
+
+memcached_return_t memcached_set_vb(memcached_st *ptr, const char *key,
```

Comment:
Nope, you've been tricked by the weird API, you provide the key twice (once as the "group key" and one as the "key"... I'm not clear on the distinction).